doc/source/reference/api/datasets/hub.rst

```diff
@@ -0,0 +1,161 @@
+It offers functionality for discovering available datasets, loading datasets, and working
+with custom dataset configurations.
+
+Dataset Registration and Asset Cards
+------------------------------------
+
+Datasets in fairseq2 can be registered in two ways:
+
+1. **Through Dataset Families**:
+    Use ``register_dataset_family`` to register custom datasets:
+
+    .. code-block:: python
+
+         from fairseq2.composition import register_dataset_family
+
+        register_dataset_family(
+            container,             # DependencyContainer instance
+            "custom_dataset",      # family name
+            CustomDataset,         # dataset class
+            CustomDatasetConfig,   # config class
+            opener=custom_opener   # optional opener function
+        )
+
+2. **Through Asset Cards**:
+    Create YAML files describing your datasets in any of these locations:
+
+    - Built-in: ``fairseq2/assets/cards/``
+    - System-wide: ``/etc/fairseq2/assets/``
+    - User-specific: ``~/.config/fairseq2/assets/``
+    - Recipe-local: Add to recipe config common section via ``extra_paths``:
+
+    .. code-block:: yaml
+
+        # Example: Adding recipe-local asset paths in config (absolute path)
+        common:
+            assets:
+                extra_paths: ["/path/to/assets"]
+        
+        # or relative path
+        # e.g. if the common section is in /path/to/recipe/config.yaml
+        # it recursively retrieves assets from /path/to/recipe/assets
+        common:
+            assets:
+                extra_paths: ["${dir}/assets"]
+
+Core Classes
+------------
+
+DatasetHub
+~~~~~~~~~~
+
+.. autoclass:: DatasetHub
+    :members:
+    :show-inheritance:
+
+    The main hub class for managing datasets. Provides methods for:
+
+    - Listing available datasets (:meth:`iter_cards`)
+    - Loading dataset configurations (:meth:`get_dataset_config`)
+    - Opening datasets (:meth:`open_dataset`)
+    - Opening custom datasets (:meth:`open_custom_dataset`)
+
+    Example usage:
+
+    .. code-block:: python
+
+        get_my_dataset_hub = DatasetHubAccessor(
+            MY_DATA_FAMILY_NAME, kls=MyDataset, config_kls=MyDatasetConfig
+        )
+
+        # Get the dataset hub
+        hub = get_my_dataset_hub()
+
+        # List all available datasets
+        for card in hub.iter_cards():
+            print(f"Found dataset: {card.name}")
+
+        # Load a dataset configuration
+        config = hub.get_dataset_config("my_dataset")
+
+        # Open a dataset
+        dataset = hub.open_dataset("my_dataset")
+
+        # Open a custom dataset with specific configuration
+        custom_dataset = hub.open_custom_dataset(config)
+
+DatasetHubAccessor
+~~~~~~~~~~~~~~~~~~
+
+.. autoclass:: DatasetHubAccessor
+    :members:
+    :show-inheritance:
+
+    Factory class for creating :class:`DatasetHub` instances for specific dataset families.
+    Used internally by model-specific hub accessors.
```

Comment:
Maybe instead of saying "internally", we should say that the direct use of `DatasetHubAccessor` (similarly model and tokenizer hub accessors) is mostly relevant for dataset/tokenizer/model implementors. It is by no means internal and has to be used in fairseq2 code base only.